hydra-node/src/Hydra/Node.hs

```diff
@@ -128,7 +129,9 @@ stepHydraNode ::
 stepHydraNode tracer node = do
   e <- nextEvent eq
   traceWith tracer $ BeginEvent party e
-  atomically (processNextEvent node e) >>= \case
+  outcome <- atomically (processNextEvent node e)
+  traceWith tracer (LogicOutcome party outcome)
+  case outcome of
```

Comment:
REVIEW: we already trace every outcome case below.

Reply:
The Error is duplicate now, yes. Maybe we can remove tracing `ErrorHandlingEvent` as we already would see it by the `LogicOutcome` trace?

Reply:
Also: we could move the `traceWith tracer (EndEvent party e)` to line 145 (at the end of the `stepHydraNode` function's `do` block.